MAS/MAS.py

```diff
@@ -832,8 +832,20 @@ def Analyse_RowAs_Kmeans(self):
         print("mean_nb_RALs", means_grp)
         
         if (p_value < 0.0001):
+
+            # Find the group with the minimum mean
+            min_mean_group_index = np.argmin(means_grp)
+
+            _indeces_grp0 = np.array(_indeces_grp0)
+            _indeces_grp1 = np.array(_indeces_grp1)
+
+            # Get the indices associated with the group with the minimum mean
+            index_small_grp = list((_indeces_grp0, _indeces_grp1))[min_mean_group_index]
+
+            _indeces_grp0 = tuple(_indeces_grp0)
+            _indeces_grp1 = tuple(_indeces_grp1)
             
-            index_small_grp = list(np.array((_indeces_grp0,_indeces_grp1))[np.where(means_grp == min(means_grp))][0][0])
+            #index_small_grp = list(np.array((_indeces_grp0,_indeces_grp1))[np.where(means_grp == min(means_grp))][0][0])
```

Comment:
I agree the original way this was coded is super wacky. I think you went in the good direction to simplify it.
Here is what really worked for me:

`index_small_grp = np.ravel((_indeces_grp0, _indeces_grp1)[min_mean_group_index])`

Note that I do reuse your `min_mean_group_index`